packages/checkbox/src/index.js

```diff
@@ -0,0 +1,266 @@
+        >
+          {children}
+        </div>
+      </CustomCheckboxContext.Provider>
+    );
+  }
+);
+
+////////////////////////////////////////////////////////////////////////////////
+export const CustomCheckbox = forwardRef(function CustomCheckbox(
+  {
+    checked: controlledChecked,
+    defaultChecked,
+    disabled: disabledProp,
+    checkmarks,
+    children,
+    id,
+    name,
+    onChange,
+    readOnly,
+    value,
+    ...props
+  },
+  forwardedRef
+) {
+  const [inputProps, { checked }] = useMixedCheckbox({
+    checked: controlledChecked,
+    defaultChecked,
+    disabled: disabledProp,
+    onChange,
+    readOnly
+  });
+  return (
+    <CustomCheckboxContainer
+      data-reach-custom-checkbox=""
+      {...props}
+      ref={forwardedRef}
+    >
+      {checked === true && checkmarks.true}
+      {checked === false && checkmarks.false}
+      {checked === "mixed" && checkmarks.mixed}
+      {children}
+      <MixedCheckbox
+        id={id}
+        value={value}
+        name={name}
+        readOnly={readOnly}
+        {...inputProps}
+      />
+    </CustomCheckboxContainer>
+  );
+});
+
+if (__DEV__) {
+  CustomCheckbox.propTypes = {
+    checkmarks: shape({
+      true: node,
+      false: node,
+      mixed: node
+    })
+  };
+  CustomCheckbox.displayName = "CustomCheckbox";
+}
+
+////////////////////////////////////////////////////////////////////////////////
+export const MixedCheckbox = forwardRef(function MixedCheckbox(
+  {
+    checked: controlledChecked,
+    defaultChecked,
+    disabled: disabledProp,
+    name,
+    onBlur,
+    onChange,
+    onFocus,
+    readOnly,
+    ...props
+  },
+  forwardedRef
+) {
+  const { setContainerState, visuallyHidden } = useCheckboxContext();
+  const [
+    inputProps,
+    { focused, checked, disabled, mixed, isControlled }
+  ] = useMixedCheckbox({
+    checked: controlledChecked,
+    defaultChecked,
+    disabled: disabledProp,
+    onBlur,
+    onChange,
+    onFocus,
+    readOnly
+  });
+
+  const ownRef = useRef(null);
+  const ref = useForkedRef(forwardedRef, ownRef);
+
+  useEffect(() => {
+    ownRef.current.indeterminate = mixed;
+    ownRef.current.checked = checked === true ? true : false;
+  }, [mixed, checked]);
+
+  useEffect(() => {
+    if (setContainerState) {
+      setContainerState({ checked, mixed, disabled, readOnly, focused });
+    }
+  }, [setContainerState, checked, disabled, mixed, readOnly, focused]);
+
+  if (__DEV__) {
+    checkboxErrorChecks({
+      isControlled,
+      controlledChecked,
+      defaultChecked
+    });
+  }
+
+  return (
+    <input
+      {...props}
+      {...inputProps}
+      ref={ref}
+      data-reach-mixed-checkbox=""
+      data-reach-mixed-checkbox-hidden={visuallyHidden && ""}
+      name={name}
+      readOnly={readOnly}
+      type="checkbox"
+    />
+  );
+});
+
+if (__DEV__) {
+  MixedCheckbox.propTypes = {
+    checked: oneOfType([bool, string]),
+    defaultChecked: bool,
+    disabled: bool,
```

Comment:
Do we need the rest of these prop-types? They're all just normal checkbox proptypes (and React, afaik) doesn't give us a way to extend "checkbox proptypes".

Additionally I don't think name or value need to be required. They aren't required in normal checkboxes, what makes this different?